rmw_fastrtps_shared_cpp/src/TypeSupport_impl.cpp

```diff
@@ -234,263 +250,234 @@ _create_type_name(const MembersType * members)
   std::string message_namespace(members->message_namespace_);
   std::string message_name(members->message_name_);
   if (!message_namespace.empty()) {
+    // Find and replace C namespace separator with C++, in case this is using C typesupport
+    message_namespace = rcpputils::find_and_replace(message_namespace, "__", "::");
```

Comment:
Can you explain a little bit more of why this is needed now?

Reply:
This is needed to ensure same type coming from Python or C++ is the same type at DDS X-Types level. Without this a Python application will register a different DDS-XTypes TypeObject and maybe won't communicate with a C++ application.

This rename is already done in other places like [MessageTypeSupport_impl](https://github.com/ros2/rmw_fastrtps/blob/rolling/rmw_fastrtps_dynamic_cpp/src/MessageTypeSupport_impl.hpp#L47).